src/Handler/Customer/SendResetPasswordTokenHandler.php

```
@@ -31,10 +31,12 @@ public function __invoke(SendResetPasswordToken $resendResetPasswordToken): void
 
         /** @var ShopUserInterface $user */
         $user = $this->userRepository->findOneByEmail($email);
-
-        Assert::notNull($user, sprintf('User with %s email has not been found.', $email));
-        Assert::notNull($user->getPasswordResetToken(), sprintf('User with %s email has not verification token defined.', $email));
-
+        if (null === $user) {
+            throw new NotFoundHttpException('User with given email does not exist!');
```

Can you throw `InvalidArgumentExcpetion` (or custom exception), as handlers should be HTTP agnostic?

By design, handlers should be infrastructure agnostic because we don't know where they came from, and the most important thing for us is to keep the application in proper state.

Please, at least revert to the previous implementation

---

I created a new Custom Exception please check now